pkg/server/plugin/keymanager/awskms/fetcher.go

```diff
@@ -0,0 +1,132 @@
+			}
+
+			spireKeyID, ok := kf.spireKeyIDFromAlias(*alias.AliasName)
+			// ignore aliases/keys not belonging to this server
+			if !ok {
+				continue
+			}
+
+			a := alias
+			// trigger a goroutine to get the details of the key
+			g.Go(func() error {
+				entry, err := kf.fetchKeyEntryDetails(ctx, a, spireKeyID)
+				if err != nil {
+					return err
+				}
+
+				keyEntriesMutex.Lock()
+				keyEntries = append(keyEntries, entry)
+				keyEntriesMutex.Unlock()
+				return nil
+			})
+		}
+
+		if !paginator.HasMorePages() {
+			break
+		}
+	}
+
+	// wait for all the detail gathering routines to finish
+	if err := g.Wait(); err != nil {
+		statusErr := status.Convert(err)
+		return nil, newErrorf(statusErr.Code(), "failed to fetch aliases: %v", statusErr.Message())
+	}
+
+	return keyEntries, nil
+}
+
+func (kf *keyFetcher) fetchKeyEntryDetails(ctx context.Context, alias types.AliasListEntry, spireKeyID string) (*keyEntry, error) {
+	describeResp, err := kf.kmsClient.DescribeKey(ctx, &kms.DescribeKeyInput{KeyId: alias.AliasArn})
+	switch {
+	case err != nil:
+		return nil, newErrorf(codes.Internal, "failed to describe key: %v", err)
+	case describeResp == nil || describeResp.KeyMetadata == nil:
+		return nil, newError(codes.Internal, "malformed describe key response")
+	case describeResp.KeyMetadata.Arn == nil:
+		return nil, newErrorf(codes.Internal, "found SPIRE alias without key arn: %q", *alias.AliasArn)
+	case !describeResp.KeyMetadata.Enabled:
+		// this means something external to the plugin, deleted or disabled the key without removing the alias
+		// returning an error provides the opportunity or reverting this in KMS
+		return nil, newErrorf(codes.FailedPrecondition, "found disabled SPIRE key: %q, alias: %q", *describeResp.KeyMetadata.Arn, *alias.AliasArn)
+	}
+
+	keyType, ok := keyTypeFromKeySpec(describeResp.KeyMetadata.CustomerMasterKeySpec)
+	if !ok {
+		return nil, newErrorf(codes.Internal, "unsupported key spec: %v", describeResp.KeyMetadata.CustomerMasterKeySpec)
+	}
+
+	publicKeyResp, err := kf.kmsClient.GetPublicKey(ctx, &kms.GetPublicKeyInput{KeyId: alias.AliasArn})
+	switch {
+	case err != nil:
+		return nil, newErrorf(codes.Internal, "failed to get public key: %v", err)
+	case publicKeyResp == nil || publicKeyResp.PublicKey == nil || len(publicKeyResp.PublicKey) == 0:
+		return nil, newError(codes.Internal, "malformed get public key response")
+	}
+
+	return &keyEntry{
+		Arn:       *describeResp.KeyMetadata.Arn,
+		AliasName: *alias.AliasName,
+		PublicKey: &keymanager.PublicKey{
+			Id:       spireKeyID,
+			Type:     keyType,
+			PkixData: publicKeyResp.PublicKey,
+		},
+	}, nil
+}
+
+func (kf *keyFetcher) spireKeyIDFromAlias(aliasName string) (string, bool) {
+	trustDomain := sanitizeTrustDomain(kf.trustDomain)
+	prefix := path.Join(aliasPrefix, trustDomain, kf.serverID) + "/"
+	if !strings.HasPrefix(aliasName, prefix) {
+		return "", false
+	}
+	return strings.TrimPrefix(aliasName, prefix), true
```

Comment:
nit: if you need the trimmed result anyway, the result of TrimPrefix can be used to infer if it had the prefix:
```
trimmed := strings.TrimPrefix(aliasName, prefix)
if trimmed == aliasName {
    return "", false
}
return trimmed, true
``` 
```